Firestore/core/src/api/stages.cc

```diff
@@ -304,6 +306,133 @@ google_firestore_v1_Pipeline_Stage RemoveFieldsStage::to_proto() const {
   return result;
 }
 
+google_firestore_v1_Pipeline_Stage ReplaceWith::to_proto() const {
+  google_firestore_v1_Pipeline_Stage result;
+  result.name = nanopb::MakeBytesArray("replace_with");
+
+  result.args_count = 1;
+  result.args = nanopb::MakeArray<google_firestore_v1_Value>(1);
+  if (expr_) {
+    result.args[0] = expr_->to_proto();
+  } else {
+    result.args[0].which_value_type =
+        google_firestore_v1_Value_string_value_tag;
+    result.args[0].string_value = nanopb::MakeBytesArray(field_name_.value());
+  }
+
+  result.options_count = 0;
+  result.options = nullptr;
+  return result;
+}
+
+ReplaceWith::ReplaceWith(std::shared_ptr<Expr> expr)
+    : expr_(std::move(expr)), field_name_(absl::nullopt) {
+}
+ReplaceWith::ReplaceWith(std::string field_name)
+    : expr_(nullptr), field_name_(std::move(field_name)) {
```

Comment:
How does this stage work? I am not sure i understand the implementation. Is it:

1. if expr_ is set, and if it is a map, it will replace the input document?
2. if field_name_ is set, it is evaluated against the document, and if it is a map it will replace the input document?

Reply:
I think I make some mistakes here and make it overly complicated. I have change the implementation in the latest commit, which basically convert "field_name" to Expr in the Swift side.